core/storage/statistics/gae_models_test.py

```diff
@@ -43,6 +46,125 @@ def test_create_and_get_event_models(self):
+        model_instance = stat_models.StateCounterModel.get_or_create(
+            'exp_id1', 'state_name')
+        self.assertEqual(model_instance.id, 'exp_id1.state_name')
+
+
+class ExplorationAnnotationsModelTests(test_utils.GenericTestBase):
+
+    def test_create_and_get_models(self):
+        stat_models.ExplorationAnnotationsModel.create('exp_id1', '1', 5, 4, {})
+
+        model1 = stat_models.ExplorationAnnotationsModel.get('exp_id1:1')
+
+        self.assertEqual(model1.exploration_id, 'exp_id1')
+        self.assertEqual(model1.version, '1')
+        self.assertEqual(model1.num_starts, 5)
+        self.assertEqual(model1.num_completions, 4)
+        self.assertEqual(model1.state_hit_counts, {})
+
+    def test_get_versions(self):
+        stat_models.ExplorationAnnotationsModel.create('exp_id1', '1', 5, 4, {})
+        stat_models.ExplorationAnnotationsModel.create('exp_id1', '2', 5, 4, {})
+
+        versions = stat_models.ExplorationAnnotationsModel.get_versions(
+            'exp_id1')
+
+        self.assertEqual(sorted(versions), ['1', '2'])
+
+
+class StateAnswersModelTests(test_utils.GenericTestBase):
+
+    def test_insert_submitted_answers(self):
+
+        submitted_answer_list = [{'answer': 'value'}]
+
+        stat_models.StateAnswersModel.insert_submitted_answers(
+            'exp_id', 1, 'state_name', 'interaction_id',
+            submitted_answer_list)
+
+        model1 = stat_models.StateAnswersModel.get_master_model(
+            'exp_id', 1, 'state_name')
+
+        self.assertEqual(model1.exploration_id, 'exp_id')
+        self.assertEqual(model1.exploration_version, 1)
+        self.assertEqual(model1.state_name, 'state_name')
+        self.assertEqual(model1.submitted_answer_list, submitted_answer_list)
+        self.assertEqual(model1.shard_count, 0)
+
+        # Use a smaller max answer list size so fewer answers are needed to
+        # exceed a shard. This will increase the 'shard_count'.
+        with self.swap(
+            stat_models.StateAnswersModel, '_MAX_ANSWER_LIST_BYTE_SIZE', 1):
+            stat_models.StateAnswersModel.insert_submitted_answers(
+                'exp_id', 1, 'state_name', 'interaction_id',
+                submitted_answer_list)
+            stat_models.StateAnswersModel.insert_submitted_answers(
+                'exp_id', 1, 'state_name', 'interaction_id',
+                submitted_answer_list)
+
+        # 'shard_count' will not increase as number of answers are less than
+        # the max answer list size.
+        stat_models.StateAnswersModel.insert_submitted_answers(
+            'exp_id', 1, 'state_name', 'interaction_id',
+            submitted_answer_list)
+
+        model1 = stat_models.StateAnswersModel.get_master_model(
+            'exp_id', 1, 'state_name')
+
+        self.assertEqual(model1.shard_count, 2)
+
+
+class StateAnswersCalcOutputModelTests(test_utils.GenericTestBase):
+
+    def test_create_and_get_models(self):
```

Comment:
Same comments as above, test non-happy-paths too. Also have clearer names for tests.

Reply:
Done but the test name does imply what we are trying to do -- create the model and get its properties.

Reply:
Avoid 'and' in test names, tests should be focusing on a single issue: `test_create_**then**_get_models`.

Reply:
"get its properties" is not part of the test name.

Prefer something like `test_get_model_returns_created_properties`.

Reply:
Done